base/sparse/linalg.jl

```diff
@@ -499,7 +499,196 @@ function norm(A::SparseMatrixCSC,p::Real=2)
     throw(ArgumentError("invalid p-norm p=$p. Valid: 1, Inf"))
 end
 
-# TODO
+# TODO rank
+
+# cond
+function cond(A::SparseMatrixCSC, p::Real=1)
+    if p == 1
+        normA = norm(A, 1)
+        normAinv = normestinv(A)
+        return normA * normAinv
+    elseif p == Inf
+        normA = norm(A, Inf)
+        normAinv = normestinv(A')
+        return normA * normAinv
+    elseif p == 2
+        throw(ArgumentError("2-norm condition number is not implemented for sparse matrices, try cond(full(A), 2) instead"))
+    else
+        throw(ArgumentError("second argment must be either 1 or Inf, got $p"))
+    end
+end
+
+function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
+
+    maxiter = 5
+
+    # Check the input
+    sizeA = size(A)
+    n = max(sizeA[1],sizeA[2])
+    F = factorize(A)
+    t = min(t,n)
+    if ~isreal(t) || length(t) > 1 || t <= 0
```

Comment:
`t` is restricted to be an `Integer` so the `isreal` check shouldn't be necessary and neither should the `length` check.